org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenProjectImporter.java

```diff
@@ -113,6 +113,19 @@ private boolean exclude(java.nio.file.Path path) {
 				}
 			}
 		}
+		if (!excluded) {
```

Comment:
Currently LocalProjectScanner used by MavenProjectImporter won't respect the `java.import.exclusions` but scan the whole workspace recursively, this is not efficient if the workspace is a polyglot project with node_modules.

Is it possible to move the exclusion logic to the scanner?

Reply:
LocalProjectScanner is within m2e. Could you create a new issue at https://bugs.eclipse.org/bugs/enter_bug.cgi?product=m2e ?